admin/server/users.go

```diff
@@ -767,3 +771,23 @@ func (s *Server) findUserAuthTokenFuzzy(ctx context.Context, input string) (*dat
 	}
 	return nil, status.Error(codes.NotFound, "token not found")
 }
+
+func resourceNamesToPB(resources []database.ResourceName) []*adminv1.ResourceName {
+	if len(resources) == 0 {
+		return nil
+	}
+	rs := make([]*adminv1.ResourceName, 0, len(resources))
+	for _, r := range resources {
+		if r.Type == "" || r.Name == "" {
+			continue
+		}
+		rs = append(rs, &adminv1.ResourceName{
+			Type: r.Type,
+			Name: r.Name,
+		})
+	}
+	if len(rs) == 0 {
+		return nil
+	}
+	return rs
+}
```

Comment:
nits, but the empty list handling here perhaps feels a bit excessive. Also, why do it need to check for `r.Type == ""`? Shouldn't that have been checked at insert time? 

I guess I would just expect:
```suggestion
func resourceNamesToPB(resources []database.ResourceName) []*adminv1.ResourceName {
	var rs []*adminv1.ResourceName
	for _, r := range resources {
		rs = append(rs, &adminv1.ResourceName{
			Type: r.Type,
			Name: r.Name,
		})
	}
	return rs
}
```